### DISCUSSION TITLE
Disable static optimization for 404 page? Custom dynamic 404 page?

### DISCUSSION THREAD

**Discussion by portenez** [vercel/next.js#15471] [Help] [answered]
Is there a good way to disable static optimization for 404 pages? Or make the 404 dynamic?

I'm trying to address this issue

```gherkin
Given I have a custom doc (_document.tsx)
And the document uses `getInitialProps`
And `getInitialProps` requires an env variable `targetUrl`
When I run `next build`
And I the env does not contain variable `targetUrl`
Then the build breaks 
```

This is because next will run the `getInitialProps` query even for the 404, however it is run during bundle/build time.

**Comment by csmartinsfct** [comment]
Did you end up sorting this out? Running into the same issue.

**Reply by portenez** [reply]
Yeah, you can opt-out of static optimization by adding a `getInitialProps` to the `_error` page. Just careful with the additional load that you'll receive. 

Added this as the answer.

**Comment by portenez** [comment, marked as answer]
You can opt-out of static optimization by adding a getInitialProps to the _error page. Just careful with the additional load that you'll receive.

Also, consider that this will also make the 5xx pages server-side rendered as opposed to static.
